### ISSUE TITLE
Maybe --apache should explain more about VirtualHosts when they're not found

### ISSUE THREAD

**Issue by schoen** [certbot/certbot#8954]
Hi!

A common problem over on the forums continues to be that people don't know that they should use VirtualHosts at all, nor how to create one, nor that Certbot expects them to create one before using `certbot --apache`.

A random current example (among several)

https://community.letsencrypt.org/t/adding-domain-to-existing-certificate/156342

Currently Certbot's output in this case will say

```
We were unable to find a vhost with a ServerName or Address of example.com.
Which virtual host would you like to choose?
```

https://github.com/certbot/certbot/blob/08839758bd80015377e8aaf61361f1c091d4c98f/certbot-apache/certbot_apache/_internal/display_ops.py#L111

In my recent experience it seems like few people who are new to Apache read this as "I should have set up my HTTP site (differently from how I did) before running Certbot" and many people instead read this as "if I choose the right thing from this menu right now, everything should will work fine, but I have no idea what to choose here".

I've been pondering various alternatives to this message that try to make clearer that the user was expected to do more to prepare the Apache environment (and should potentially abort the Certbot run and go do that other stuff first).

One idea is something like

```
Certbot's Apache integration expects your domain names to be explicitly listed in VirtualHosts in your Apache configuration. The sites corresponding to these domain names should be set up this way before Certbot is run. However, we couldn't find any existing VirtualHost on this system that mentions example.com.

If you don't have a VirtualHost for this domain name, please stop Certbot and create one first. See https://httpd.apache.org/docs/2.4/vhosts/examples.html for examples about how to do this. The best practice is to make an file in your `sites-available` directory for each site that will be served by this server, and then enable it by running "a2ensite".

If you know that an existing VirtualHost is responsible for this domain name, you can select it from the menu below and Certbot will try to use it.
```

This is probably too wordy, but it does seem to be common that people just don't get the idea that they were expected to do something here (and so are presumably using the default server for everything, which isn't really ideal for Certbot).

**Comment by osirisinferi** [comment]
*Or*, if a user has just a single website, don't require a virtualhost at all? Just let the user enter its hostname and place the SSL options in a generic configuration file, just as the Apache configuration with just HTTP was? Why require a VirtualHost for a single site?

**Comment by bmw** [comment]
Hey Seth. Thanks for continuing to think about how to improve Certbot.

My first thought is I wonder if there's anything we could do to better handle these cases automatically. If there's not the time/interest to do this work, adding better messaging is an easy fix, but at least in scenario in the post you linked, I think we probably could handle this. I imagine the sequence of events was:

1. User successfully obtained and installed a Certbot on a machine with the default vhost in the Ubuntu packages.
2. When they went to install the 2nd certificate, Certbot saw both the original HTTP vhost and the HTTPS vhost it created from the prior run and it wasn't sure which one should be modified.

Why are we confused? There's 1 HTTP vhost and 1 HTTPS vhost. Shouldn't we know which will be used depending on what we're wanting to do (e.g. install the certificate or set up the HTTP to HTTPS redirect)? Maybe there's some reason that doesn't immediately come to mind that we can't figure this out, but it seems to me like we should be able to.

If someone wants to just add better messaging though, I agree the suggested text in the initial post is a bit wordy and I have a couple other suggestions.

> Certbot's Apache integration expects your domain names to be explicitly listed in VirtualHosts in your Apache configuration...

I'm not sure this is true. I think it's fine to just let Apache fall back to the default vhost and I think Certbot handles this case automatically on fresh Debian based installations with only one vhost. I think the trouble is when there's multiple vhosts and Certbot (currently) needs help determining which vhost will be defaulted to.

> The best practice is to make an file in your `sites-available` directory for each site that will be served by this server, and then enable it by running "a2ensite".

I think this advice is Debian specific. We could maybe explicitly print this on Debian based systems which covers the majority of Certbot users, but maybe we'd want to make the instructions more general some/all of the time.
